### PR TITLE
Show all LintResult in Rule_L020

### PR DESCRIPTION
**Pull request by zhongjiajie**
### Brief summary of the change made

Rule_L020 only show first LintResult before,
after this patch merge Rule_L020 would show
all duplicate table aliases, without duplicate
position

### Are there any other side effects of this change that we should be aware of?

I think this PR no need add `std_rule_cases` here, am I wrong?

### Pull Request checklist

- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - ~`.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.`~  Already exists and no need

### CODE REVIEW

File: src/sqlfluff/core/parser/segments/base.py [hunk 1]
```diff
@@ -131,6 +131,9 @@ def __eq__(self, other):
             )
         )
 
+    def __hash__(self):
+        return hash((self.__class__.__name__, self.raw, self.pos_marker.source_position()))
+
```
Comment (zhongjiajie) [inline src/sqlfluff/core/parser/segments/base.py:136]: Add `__hash__` for `set.add` function which need hashable for remove duplicate element.

**Comment by tunetheweb** [comment]
> I think this PR no need add `std_rule_cases` here, am I wrong?

I think a test with two missing aliases are both should would be good to ensure this doesn’t regress in future.

**Comment by tunetheweb** [comment]
For more complex tests where you want to look at the actual output, you don't add a fixture, but instead add python code test/rules/. Have a look at this one for example:

https://github.com/sqlfluff/sqlfluff/blob/e21d200c35b75d771f2d7635460f9c0e7d9d6f06/test/rules/std_L007_test.py#L46-L63

**Comment by zhongjiajie** [comment]
Thanks for review, I add more test for rule L020

**Comment by zhongjiajie** [comment]
CI pass, PTAL thanks @tunetheweb

**Review by tunetheweb — APPROVED** [review]
LGTM! Nice test cases.

**Comment by zhongjiajie** [comment]
Thanks for merging.

**Pull request merged**